### ISSUE TITLE
horizon_maas plugin "“horizon_local_status”"  throws errors when using keystone domains

### ISSUE THREAD

**Issue by brainerd123** [rcbops/rpc-openstack#1262]
Hello,

When enabling Keystone Domains, the following MaaS Horizon check stops working.

"horizon_local_status"

:set consecutiveCount=3
if (metric["horizon_local_status"] != 1) {
    return new AlarmStatus(CRITICAL, "Horizon unavailable");
}

The check can no longer login to Horizon.

###### Please provide the output to 'git describe --all' for  all deployments where you have confirmed it occurs.git

root@726402-infra01:/opt/openstack-ansible# git describe --all
heads/stable/mitaka

##### What type of issue is this? Please tick the appropriate box.
- [x] bug - existing functionality that does not work for you as documented
- [ ] feature - new functionality that you would like to see added

##### What is the expected behavior?

The check should account for Domains in Horizon be able to login, which would clear the MaaS check.
##### What is actually happening?

When domains are enabled it can longer login to horizon so the check fails

##### What were you attempting to do when you encountered this error (give as much detail as possible)? 

Nothing, I just noticed the MaaS check failed when domains were enabled. 

##### How can we reproduce this issue?

Enable Domains in a Liberty or newer environment and test the MaaS check afterwards. 

##### How can we verify your issue is addressed?

Update this thread or provide me a new MaaS check to test in my environment. 

##### If you are using or know of a workaround please describe the necessary steps?

No other workarounds other than disabling the check

##### What is your evaluation on the importance of this issue, and why?

Medium importance.

----------------------------------------------------------------------------------------------------------------------------
###### For each pull request, update the branch line to reference it.
- [x] master (#1476)
- [x] mitaka-13.1 (#1488)
- [x] mitaka-13.0 (#1489)
- [x] liberty-12.2 (#1490)
- [ ] kilo

##### Upgrade impact. Consider the impact on moving between major, minor and patch releases.
- [x] This change does not impact upgrades
- [ ] This change addresses a potential upgrade impact
- [ ] The upgrade impact is being tracked by another issue.

The issue tracking the upgrade impact is:

##### Testing impact
- [ ] An issue/pull-request has been create or exists to track the testing gap, and is referenced below
- [x] This change includes additional testing
- [ ] This change is tested by the existing setup

Additional testing links:

screenshot of Horizon with Domains:

https://www.dropbox.com/s/nhsqtjwinj4tg3v/Domains.png?dl=0 

##### Release note
- [ ] This change includes a release note.
- [ ] This change does not require a release note.

##### Does your change require documentation?
- [ ] Yes, and I submitted an issue to [Docs issues](https://github.com/rackerlabs/docs-rpc/issues "Docs issues")
- [ ] Yes, and I contributed a documentation patch to [docs-rpc](https://github.com/rackerlabs/docs-rpc "docs-rpc")
- [ ] No documentation required for this change

Additional documentation links (please add links if you have ticked yes):

**Comment by Frank-ZhangXin** [comment]
@brainerd123 Hi, could you please point out where you enabled 'keystone domains'?

**Comment by brainerd123** [comment]
Hello Frank,

I believe that configuration is done here:

Enable multi-domain for Horizon

Add the following to /etc/openstack_deploy/user_variables.yml

horizon_keystone_multidomain_support: True

A line in /etc/ansible/roles/os_horizon/templates/horizon_local_settings.py.j2 must be commented out to prevent Horizon from forcing the Default domain

sed -i 's/OPENSTACK_KEYSTONE_DEFAULT_DOMAIN/#OPENSTACK_KEYSTONE_DEFAULT_DOMAIN/' /etc/ansible/roles/os_horizon/templates/horizon_local_settings.py.j2

**Comment by sigmavirus24** [comment]
@git-harry asked I drive this forward given its age.

**Comment by Frank-ZhangXin** [comment]
Status: Currently we experienced same maas fails on each backport PR, and believe this would be solved when jenkins side fixed.

**Comment by git-harry** [comment]
Blocked by https://github.com/rcbops/u-suk-dev/issues/513

**Comment by tomc603** [comment]
I've merged the available backports and checked them as complete in the original issue post. @Frank-ZhangXin Does this need to go back to Kilo still?

**Comment by sigmavirus24** [comment]
@tomc603 it seems we're not backporting things to kilo any longer. I would guess it doesn't need to go to kilo.

**Comment by brainerd123** [comment]
Hey Guys,

No need to backport.  Please merge into Mitaka.  Thanks for addressing this.

AB

**Comment by tomc603** [comment]
Given there's no need to backport to Kilo, then I'd say @Frank-ZhangXin could potentially move this issue to done. Unless we want docs updates or release notes.
